dashboard/app/models/user.rb

```diff
@@ -2764,6 +2766,26 @@ def should_validate_us_state?
     new_record? || us_state_changed?
   end
 
+  def student_in_lockout_flow?
+    return unless user_type == TYPE_STUDENT
+    return unless %w[US RD].include? country_code
+    return unless birthday_changed? || us_state_changed?
```

Comment:
I think it might be good to move these returns to their own function, perhaps, something like `should_check_age_or_state_update?` or similar to separate concerns. It's debatable. It's nice to know that a function that is called for every `User` update is not likely to run... and a good way to do that is separate the concern by putting the rule for when it runs or not into its own place and _that_ gets called every update, but better guarantees that the function doing the destructive work is limited.